scripts/hail_batch/snp_chip_generate_pca/snp_chip_generate_pca.py

```diff
@@ -0,0 +1,46 @@
+"""
+Generate PCA on TOB-WGS data, filtered
+to variants present in SNP-chip data.
+"""
+
+import click
+import hail as hl
+import pandas as pd
+
+TOB_WGS = 'gs://cpg-tob-wgs-main/mt/v2-raw.mt/'
+SNP_CHIP = 'gs://cpg-tob-wgs-test/snpchip/v1/snpchip_grch38.mt/'
+
+
+@click.command()
+@click.option('--output', help='GCS output path', required=True)
+def query(output):  # pylint: disable=too-many-locals
+    """Query script entry point."""
+
+    hl.init(default_reference='GRCh38')
+
+    tob_wgs = hl.read_matrix_table(TOB_WGS).key_rows_by('locus', 'alleles')
+    snp_chip = hl.read_matrix_table(SNP_CHIP).key_rows_by('locus', 'alleles')
+
+    # filter to loci that are contained in snp-chip data after densifying
+    tob_wgs = hl.experimental.densify(tob_wgs)
+    tob_wgs = tob_wgs.semi_join_rows(snp_chip.rows())
+    tob_wgs = tob_wgs.cache()
+    print(tob_wgs.count_rows())
+    tob_wgs_path = f'{output}/tob_wgs_filtered_snp_chip.mt'
+    tob_wgs = tob_wgs.repartition(100, shuffle=False)
+    tob_wgs.write(tob_wgs_path)
```

Comment:
How about forming the `union` of sample columns? That'll give you the row restriction immediately and you can run a joint PCA afterwards.

Reply:
Would it be better to project the SNP-chip samples onto the tob-wgs pca space? I guess the question is whether we want to look at the SNP-chip data in relation to the tob-wgs sample mean, or if we want to allow the snp-chip samples to skew the mean. I think the datasets should be similar enough that it won't actually matter either way, but my first thought was projecting the snp-chip data back onto the tob-wgs pca space

Reply:
Wasn't that what you ran previously? If you use loci that aren't present in the SNP-chip data, I think you'll always get the "shrinkage" effect that your previous plot showed.

Reply:
The previous plot was without any filtering. This is filtering the tob-wgs matrix table to variants that are present in the snp-chip data (and I would also have to filter the snp-chip data to variants that are present in the tob-wgs matrix table). I'm happy to do a union of the sample columns, if that seems more efficient. I don't think the effect will be huge either way